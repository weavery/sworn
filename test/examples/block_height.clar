(define-data-var height uint 0)

(define-read-only (read)
  (ok (var-get height)))

(define-public (update)
  (begin
    (var-set height block-height)
    (ok (var-get height))))
