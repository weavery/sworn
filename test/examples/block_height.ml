(* This is free and unencumbered software released into the public domain. *)

#include "tests.ml"

let () = compile_and_print "block_height.clar"
