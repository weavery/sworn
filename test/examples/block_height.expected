// Generated by Sworn

clarity.requireVersion("0.0")

function read(state) {
  let result = null
  result = clarity.ok(state.height)
  return result
}

function update(state) {
  let result = null
  state.height = clarity.blockHeight()
  result = clarity.ok(state.height)
  return state
}

export function handle(state, action) {
  const input = action.input
  if (input.function === 'read') {
    return {result: read(state)}
  }
  if (input.function === 'update') {
    return {state: update(state)}
  }
  return {state}
}
